### DISCUSSION TITLE
Help - Publish API

### DISCUSSION THREAD

**Discussion by Aiko0708** [apiman/apiman#2630] [Help]
Hello,

I'm trying to configure apiman 3.1.2, but I have a strange error.

When I try to publish an api, at the manager level, I am redirected to the page /apimanui/api-manager/errors/500: "Server Error!
Unexpected Server Error. Please try again." "Failed to published API".

However at the elasticsearch database level, I see well the API published on the gateway but on the manager it remains in "ready" status.

On the manager, here is what I have as logs:

WARN [io.apiman.manager.api.war.wildfly8.ApimanTransactionalInterceptor] (default task-5) Transaction has been set as rollback only because of io.apiman.manager.api.rest.exceptions.ActionException: Failed to publish API. (This can occur accidentally if you are using exceptions for control-flow, which can caused the TX to be marked rollback only)
16:04:52,812 WARN [io.apiman.manager.api.war.wildfly8.ApimanTransactionalInterceptor] (default task-5) Transaction has been set as rollback only because of io.apiman.manager.api.rest.exceptions.ActionException: Failed to publish API.. (This can occur accidentally if you are using exceptions for control-flow, which can caused the TX to be marked rollback only)
16:04:52,814 ERROR [io.apiman.manager.api.rest.exceptions.mappers.RestExceptionMapper] (default task-5) Failed to publish API.: io.apiman.manager.api.rest.exceptions.ActionException: Failed to publish API.
..................................
Caused by: io.apiman.gateway.engine.beans.exceptions.PublishingException: Error publishing API.

Has anyone encountered the same error before?

Thank you for your help

**Comment by msavy** [comment]
If you think there's a bug, please fill in a bug report
